R/add_creator_id.R

```diff
@@ -5,32 +5,49 @@
 #' @param orcid ORCID in the format 'https://orcid.org/WWWW-XXXX-YYYY-ZZZZ'
 #' @param id reference ID. Character string to set reference ID for creators with additional roles (i.e. metadataProvider, etc.)
 #' @param surname creator surname (last name), defaults to first creator if not specified. Not case-sensitive.
+#'
 #' @keywords eml creator orcid id
+#'
 #' @details All prameters other than the EML are optional, but since the point of the function is to modify either the orcid, ref id, or both, you better specify at least one. Requires the crayon package.
+#'
 #' @export
+#'
 #' @examples
-#' add_creator_id(eml2, orcid = "https://orcid.org/WWWW-XXXX-YYYY-ZZZZ")
+#' library(dataone)
+#' library(arcticdatautils)
+#' library(EML)
 #'
+#' cnTest <- dataone::CNode('STAGING')
+#' mnTest <- dataone::getMNode(cnTest,'urn:node:mnTestARCTIC')
+#' eml_pid <- arcticdatautils::create_dummy_metadata(mnTest)
+#' eml1 <- EML::read_eml(rawToChar(getObject(mnTest, eml_pid)))
+#' add_creator_id(eml1, orcid = "https://orcid.org/WWWW-XXXX-YYYY-ZZZZ")
 
-#function for adding orcid or reference id to creator
 add_creator_id <- function(eml,
                            orcid = NULL,
                            id = NULL,
                            surname = NULL) { #not sensitive to capitalization
     library(crayon)
 
+    #variable checks:
+    for (args in c(orcid, id, surname)) {
+        if (!(is.null(args) | is.character(args))) {
+            stop(paste(args, "must be a character string."))
+        }
+    }
+
     #grab creators and save to new variable
     creatorList <- eml@dataset@creator@.Data
 
     #determine surname position to access correct creator
     #if none are specified, the first creator will be modified
-    if(is.null(surname == TRUE)){
+    if (is.null(surname)) {
         cat(green("Since surname was not specified, the first creator entry will be modified. "))
         pos <- 1
     } else {
         #make vector of creator surnames
         surNames <- rep(NA, times = length(creatorList))
-        for(i in 1:length(creatorList)){
+        for (i in 1:length(creatorList)) {
             creator1 <- creatorList[[i]]
             surName1 <- creator1@individualName@.Data[[1]]@surName@.Data
```

Comment:
Should be
```
surName1 <- creator1@individualName[[1]]@surName@.Data
```
But this will only get the first individualName listed, there could be more than 1

Reply:
Both seem to work.  Not sure what the preferred way is but I'll check into it. (which might mean something else went wrong when I tried your previous suggestion along these lines)

It only returns one individualName since you're only looking at a single creator entry

Reply:
Update: I guess there's no difference!  I'll delete the .Data to make it cleaner.